docker-compose.yml

```diff
@@ -0,0 +1,14 @@
+version: '3.2'
+services:
+  pl:
+    build:
+      context: .
+    ports:
+      - 3000:3000
+    volumes:
+      - ./testCourse:/course
+      - /var/run/docker.sock:/var/run/docker.sock
+      - ${HOME}/pl_ag_jobs:/jobs
+    container_name: pl
+    environment:
+      - HOST_JOBS_DIR=${HOME}/pl_ag_jobs
```

Comment:
Does this replace the huge `docker run` command with something like `docker-compose -p pl up`?

Reply:
This should. If you just run `docker-compose up`, then the entire stack should go up, with the container named `pl`.

Before this comes along for the ride on my PR (or i pull it out into another PR), I did want to check if this had all the stuff we actually need to run PL in a docker container with someone else from the team.

Reply:
This is awesome! Is there a way to replicate something like this where we append a shell command:
`docker run ... prairielearn/prairielearn /PrairieLearn/docker/start_workspace.sh`

Reply:
> or i pull it out into another PR

Also imo this is worth splitting out into its own PR.

Reply:
Yup. You can either do:
`docker-compose run /PrairieLearn/docker/start_workspace.sh` (one time command in a running container)
or, alternatively, feed it in as a [command to the container](https://docs.docker.com/compose/compose-file/#command), overriding the `/PrairieLearn/docker/init.sh
` at the end of the Dockerfile.
```yml
version: '3.2'
services:
  pl:
    command: /PrairieLearn/docker/start_workspace.sh
```
We could add this as a commented out line in the docker compose, so people can play with it as they need. If it's accidently committed, then we can just remind someone to comment it back out.

I'm not 100% sure what `start_workspace.sh` does--is it a wrapper of `init.sh`? Does it spin up other services or containers?

Since this sounds like something that's more generally useful (and this is really a draft-draft for manual grading rn), should I pull this out into a separate PR and request you and matt on it? We can always add more stuff to the file after, but this might be a good first step to kill the magical bash script floating around.

Reply:
> I'm not 100% sure what start_workspace.sh does--is it a wrapper of init.sh? Does it spin up other services or containers?

Yeah it's just a wrapper that splits the main PL server and workspace host server into separate tmux panes. The main `init` script lumps output from both servers together, but when dev'ing workspaces, it's useful to view the servers separately.

Reply:
Out of curiosity, do we run these as separate servers in prod.

Reply:
Yeah they are separate. Based on https://github.com/PrairieLearn/PrairieLearn/issues/3178#issuecomment-698030570, it looks like these are also separate:

> * Main PL server
> * Chunk servers
> * PrairieGrader hosts
> * Workspace hosts

Reply:
If they're separate servers, it may be worth looking into pulling them apart into different services (i.e. containers) in the compose file.

For now though, I'll shift the basic docker compose to a different pr and update docs, making notes about running a different entry command, and commenting out the command line for now.